lib/Status.php

```diff
@@ -29,24 +29,31 @@ class Status {
 	 *  The file was checked and found to be infected.
 	 */
 	const SCANRESULT_INFECTED = 1;
-	
+
 	/*
 	 * Should be SCANRESULT_UNCHECKED | SCANRESULT_INFECTED | SCANRESULT_CLEAN
 	 */
 	protected $numericStatus;
-	
+
 	/*
 	 * Virus name or error message
 	 */
 	protected $details = "";
-	
+
 	protected $ruleMapper;
-	
+
 	public function __construct() {
 		$this->numericStatus = self::SCANRESULT_UNCHECKED;
 		$this->ruleMapper = new Db\RuleMapper(\OC::$server->getDb());
 	}
-	
+
+	public static function create(int $status, string $details = '') {
```

Comment:
I guess the expected usage for the class is to create the `Status` object, and then run `$status->parseResponse(...)`. This is intended to set the proper status and details based on the output response of the virus scanner. In this regard, this new `create` method is risky because you could bypass those checks.
Note that the method is currently unused within this repo.

My main worry is that, after half a year we think: 1) the method is risky because it can be misused, and 2) the method isn't used. As a consequence, we remove the method, and then we break an app that we hadn't considered.
If the method is required for a 3rd-party app, we need to make sure that dependency is visible, specially when the files_antivirus doesn't have an API for this matter.